File: 1-10/3-Longest-Substring-Without-Repeating-Characters/C++/Program.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include<algorithm>
using namespace std;

class Solution
{
public:
    int lengthOfLongestSubstring(string s)
    {
        if (s.length() == 0)
        {
            return 0;
        }

        vector<int> num;
        vector<char> str = {s[0]};
        int size = s.length();
        int j = 0;
        for (int i = 1; i < size; i++)
        {
            if (!count(str.begin(), str.end(), s[i]))
                str.push_back(s[i]);
            else
            {
                num.push_back(str.size());
                while (count(str.begin(), str.end(), s[i]))
                    str.erase(str.begin());
                str.push_back(s[i]);
            }
        }
        num.push_back(str.size());
        return *max_element(num.begin(), num.end());
    }
};

int main(int argc, char *argv[])
{
    Solution s;
    string test = "pwwqasw";
    cout << s.lengthOfLongestSubstring(test);
    return 0;
}

File: 1-10/3-Longest-Substring-Without-Repeating-Characters/Python/program.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        if len(s) == 0:
            return 0

        strList = []
        strList.append(s[0])
        numList = []        
        
        for i in range(1, len(s)):
            if s[i] not in strList:
                strList.append(s[i])
            else:
                numList.append(len(strList))
                j = 0
                while s[i] in strList:
                    strList.pop(j)
                strList.append(s[i])

        numList.append(len(strList))
        return max(numList)

#Test
test = "pwwqasw"
s = Solution()
print(s.lengthOfLongestSubstring(test))